js/admission.js

```diff
@@ -0,0 +1,134 @@
+            <div class="resultcard-content-col">
+            <p class="line-spacing"><span class="subtitle">Period: </span>${course.period}</p>
+            <p class="line-spacing"><span class="subtitle">Level: </span>${course.level}</p>
+            <p class="line-spacing"><span class="subtitle">Subject Area: </span>${course.subjectArea}</p>
+            </div>
+            <div class="resultcard-content-col">
+            <p class="line-spacing"><span class="subtitle">Pace of study: </span>${course.paceOfStudy}</p>
+            <p class="line-spacing"><span class="subtitle">Teaching form: </span>${course.teachingForm}</p>
+            <p class="line-spacing"><span class="subtitle">Application code: </span>${course.applicationCode}</p>
+            </div>
+            </div>
+            </div>`;
+            $("#search-result").append(html);
+        });
+    }
+
+    function displayerrorMessage(){
+        var html = `<div class="noMatchingResult searchresult-card">
+                            <div class="image-container"></div>
+                            <p class="subtitle text-center">Your search criteria did not match any courses or programs</p>
+                            </div>`;
+                            $("#search-result").append(html);
+    }
+
+// display result as per searched text and semester selection in search bar
+
+    var filteredList = [];
+    courseDataAfterSearch =  courseData;
+    $("#search").submit(function() {
+        var searchText = $("#freeText").val();
+        var semester = $("#period").val();
+        filteredList = courseData.filter(course => ((course.period.toUpperCase().indexOf(semester.toUpperCase()) != -1) && (course.subjectArea.toUpperCase().indexOf(searchText.toUpperCase()) != -1)));
+        populateUI(filteredList);
+        courseDataAfterSearch = filteredList;
+        resetFilters();
+        event.preventDefault();
+    });
+    
+    //filter serach results as per selected filters in the filter bar
+
+    $('#search-form').change(filterData);
+    function filterData () {
+        var checkedFilters = getSelectedFilters();
+        checkedCheckbox = checkedFilters.filter(filter => (filter.type ==="checkbox"));
+        checkedRadio = checkedFilters.filter(filter => (filter.type==="radio"));
+        checkedDropdown = checkedFilters.filter(filter => (filter.type ==="dropdown"));
+        final_filtered_list = courseDataAfterSearch;
+
+        //if radio button is selected, update filtered list as per that filter
+
+        if  ((checkedRadio.length !=0) && (checkedRadio[0].value)!="both"){
+            final_filtered_list = final_filtered_list.filter(course => (course.teachingForm.toUpperCase().indexOf(checkedRadio[0].value.toUpperCase()) != -1));
+        }
+
+        //if checkbox is selected, update filtered list as per that filter
+
+        if (checkedCheckbox.length!=0) {
+            var filteredListForAllCheckBoxFilters = [];
+            for(var i in checkedCheckbox) {
```

Comment:
you can use `let` here as long as you don't need this variable in the file

also, `i` name doesn't give any information about data you process, it's better to replace with something meaningful